### PR TITLE
Extend Linux CI support

### PR DESCRIPTION
**Pull request by jaredpar**
This gets us one step closer to running unit test in our Jenkins system.

Most of this change is simply fixing path casing issues in our build files.

### CODE REVIEW

**Comment by jaredpar** [comment]
@tmat, @VSadov, @agocke, @gafter, @AlekseyTs

**Comment by gafter** [comment]
LGTM

**Comment by jaredpar** [comment]
@dotnet-bot  test this please

File: src/Tools/Microsoft.CodeAnalysis.Toolset.Open/Targets/GenerateAssemblyInfo.targets [hunk 1]
```diff
@@ -64,7 +64,7 @@
           Inputs="$(MSBuildThisFileFullPath)"
           Outputs="$(GeneratedAssemblyInfoFile)"
           DependsOnTargets="PrepareForBuild"
-          Condition="'$(Language)' == 'C#' or '$(Language)' == 'VB'">
+          Condition="('$(Language)' == 'C#' or '$(Language)' == 'VB') and '$(OS)' == 'Windows_NT' ">
 
     <WriteCodeFragment AssemblyAttributes="@(AssemblyVersionAttribute)"
                        Language="$(Language)"
```
Comment (VSadov) [inline src/Tools/Microsoft.CodeAnalysis.Toolset.Open/Targets/GenerateAssemblyInfo.targets]: Is this not yet supported or completely unnecessary outside NT ?

**Comment by VSadov** [comment]
LGTM

**Comment by AlekseyTs** [comment]
LGTM

File: src/Tools/Microsoft.CodeAnalysis.Toolset.Open/Targets/GenerateAssemblyInfo.targets (hunk 1 above)
Comment (jaredpar) [inline src/Tools/Microsoft.CodeAnalysis.Toolset.Open/Targets/GenerateAssemblyInfo.targets]: There is a bug in the mono implementation of the VB CodeDom Code Generator that causes this to not compile.  I have a patch in hand to fix it but until I can get the necessary approval for that patch I just disabled this portion of the build.

**Pull request merged**